src/libs/ajax/Catalog.ts

```diff
@@ -0,0 +1,169 @@
+import * as _ from 'lodash/fp'
+import { authOpts, fetchCatalog, jsonBody } from 'src/libs/ajax/ajax-common'
+
+
+interface DatasetTableListEntry {
+  name: string
+  hasData?: boolean
+}
+
+interface DatasetTableListResponse {
+  tables: DatasetTableListEntry[]
+}
+
+interface Column {
+  name?: string
+}
+
+interface DatasetTableResponse {
+  columns: Column[]
+  rows: any[]
+}
+
+
+type DataUsePermission =
+    'DUO:0000007' |
+    'DUO:0000042' |
+    'DUO:0000006' |
+    'DUO:0000011' |
+    'DUO:0000004'
+
+type CloudRegion =
+    'southamerica-west1' |
+    'us-central1' |
+    'us-east1' |
+    'us-east4' |
+    'us-west1' |
+    'us-west4' |
+    'europe-north1' |
+    'europe-west1' |
+    'europe-west4' |
+    'asia-east1' |
+    'asia-southeast1'
+
+type CloudResource =
+    'bigquery' |
+    'firestore' |
+    'bucket'
+
+type CloudPlatform =
+    'gcp' |
+    'azure'
+
+type AccessLevel =
+    'owner' |
+    'reader' |
+    'discoverer' |
+    'no_access'
+
+interface Publication {
+  'dct:title'?: string
+  'dcat:accessURL'?: string
+}
+
+interface DataCollection {
```

Comment:
Hmm.... I am seeing quite a few optional? properties in these types... Is this a side-effect of using that TS-from-API generation tool you mentioned?

Reply:
They are truly actually optional in the schema. I didn't end up using a library, but rather copied the signature of the types that we have here: https://github.com/DataBiosphere/terra-data-catalog/blob/main/service/src/main/resources/schema/development/schema.json

Reply:
ok cool